packages/wcg-url/index.ios.ts

```diff
@@ -0,0 +1,199 @@
+interface BlobItem {
+  blob: Blob;
+  path?: string;
+  type?: string;
+  ext?: string;
+}
+
+const BLOB_STORE = new Map<string, BlobItem>();
+
+import './url-search';
+
+export class WCGURL implements URL {
+  private _native: NSCWCGURL;
+  _isBlobURL = false;
+
+  constructor(url: string, base?: string | URL) {
+    if (url?.startsWith?.('blob:')) {
+      this._isBlobURL = true;
+    }
+    let baseUrl: string;
+    if (typeof url === 'string' && url.startsWith('blob:')) {
+      this._native = NSCWCGURL.alloc().initWithValueBase(url, null);
+    } else {
+      if (base instanceof WCGURL) {
+        baseUrl = base._native.toString();
+      } else if (base) {
+        try {
+          baseUrl = base.toString();
+        } catch (e) {
+          throw new TypeError(`Failed to construct 'URL': Invalid base URL`);
+        }
+      }
+      try {
+        if (baseUrl) {
+          this._native = NSCWCGURL.alloc().initWithValueBase(url, baseUrl);
+        } else {
+          this._native = NSCWCGURL.alloc().initWithValueBase(url, null);
+        }
```

Comment:
```suggestion
        this._native = NSCWCGURL.alloc().initWithValueBase(url, baseUrl ?? null);
```
(Or `||` if you wanna exclude empty-string too)